packages/blade/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { isUndefined } from 'lodash';
+import React from 'react';
+import { isEmpty } from '../../utils';
+import { useCheckboxGroupContext } from './CheckboxGroup/CheckboxGroupContext';
+import { CheckboxHintText } from './CheckboxHintText';
+import { CheckboxIcon } from './CheckboxIcon';
+import { CheckboxInput } from './CheckboxInput';
+import { CheckboxLabel } from './CheckboxLabel';
+import { CheckboxLabelText } from './CheckboxLabelText';
+import { useCheckbox } from './useCheckbox';
+import { Wrapper } from './Wrapper';
+
+type CheckboxProps = {
+  /**
+   * If `true`, sets the checkbox will be checked.
+   * Use `onChange` to update its value
+   *
+   * @default false
+   */
+  isChecked?: boolean;
+  /**
+   * If `true`, the checkbox will be initially checked.
+   *
+   * @default false
+   */
+  defaultChecked?: boolean;
+  /**
+   * The callback invoked when the checked state of the `Checkbox` changes.
+   */
+  onChange?: (isChecked: boolean) => void;
```

Comment:
the api changes aren't propagated here?

Reply:
Right will update.

Reply:
Wanted to discuss. should we make it an object or just naked parameters? 

Says because users will 90% of the time will only case about the isChecked in that case it looks kinda ugly.

```ts
<Checkbox onChange={( { isChecked } ) => setChecked(isChecked} />

// vs

<Checkbox onChange={(isChecked) => setChecked(isChecked} />
```

Reply:
Done. updated with {isChecked,event,value}